docs/Adding-Software-to-Homebrew.md

```diff
@@ -8,46 +8,139 @@ Before you start, please check the open pull requests for [Homebrew/homebrew-cor
+
+1. Run `brew audit --formula --new <formula>` with your formula. If any errors occur, correct your formula and run the audit again. The audit should finish without any errors by the end of this step.
+
+1. Run `brew style --formula --fix <formula>` to automatically check and correct your formula's conformity to Homebrew's style guidelines.
 
 1. Run your formula's test using `brew test <formula>`. The test should finish without any errors.
 
+Keep in mind that these checks will happen automatically when you submit your pull request. Completing them beforehand not only saves time but also makes the whole process smoother for everyone.
+
+If your application and Homebrew do not work well together, feel free to [file an issue](https://github.com/Homebrew/homebrew-core/issues/new/choose) after checking out open issues.
+
 ### Submitting the formula
 
-You're finally ready to submit your formula to the [homebrew-core](https://github.com/Homebrew/homebrew-core) repository. If you haven't done this before, you can refer to the [How to Open a Homebrew Pull Request](How-To-Open-a-Homebrew-Pull-Request.md#formulae-related-pull-request) documentation for help. Maintainers will review the pull request and provide feedback about any areas that need to be addressed before the formula can be added to Homebrew.
+You're now ready to contribute your formula to the [homebrew-core](https://github.com/Homebrew/homebrew-core) repository. If this is your first time making a submission, consult the [How to Open a Homebrew Pull Request](How-To-Open-a-Homebrew-Pull-Request.md#formulae-related-pull-request) guide for detailed instructions. Your pull request will be reviewed by the maintainers, who may suggest improvements or changes before your formula can be added to Homebrew.
 
-If you've made it this far, congratulations on submitting a Homebrew formula! We appreciate the hard work you put into this and you can take satisfaction in knowing that your work may benefit other Homebrew users as well.
+If you've made it this far, congratulations on successfully submitting a Homebrew formula! Your dedication and hard work are highly valued. Take satisfaction in knowing that your contribution provides a valuable addition that will benefit many Homebrew users.
 
 ## Casks
 
 **Note:** Before taking the time to craft a new cask:
 
-* make sure it can be accepted by checking the [Rejected Casks FAQ](Acceptable-Casks.md#rejected-casks), and
+* make sure it can be accepted by checking [Acceptable Casks](Acceptable-Casks.md).
 * check that the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
+* if you are just updating the version of an existing cask, see the streamlined method in ['submit a new version of an existing cask'](How-To-Open-a-Homebrew-Pull-Request.md#submit-a-new-version-of-an-existing-cask).
```

Comment:
I appreciate this was already there but, as above, this feels a bit redundant given the last non-formula section before.

Reply:
See my response to the other case of this here:

- https://github.com/Homebrew/brew/pull/16327#discussion_r1425287567